lib/Reader/reader.cpp

```diff
@@ -8625,3 +8662,166 @@ void ReaderBase::resolveToken(mdToken Token, CorInfoTokenKind TokenType,
+    default:
+      assert(UNREACHED);
+    }
+    if (opResult) {
+      return opResult;
+    }
+  }
+
+  ReaderOperandStack->push(Arg1);
+  ReaderOperandStack->push(Arg2);
+  return 0;
+}
+
+IRNode *ReaderBase::generateSIMDUnOp(ReaderSIMDIntrinsic OperationCode) {
+  IRNode *Arg = ReaderOperandStack->pop();
+  IRNode *Vector = Arg;
+  if (Vector) {
+    IRNode *opResult = 0;
+    switch (OperationCode) {
+    case ABS:
+      opResult = vectorAbs(Vector);
+      break;
+    case SQRT:
+      opResult = vectorSqrt(Vector);
+      break;
+    default:
+      assert(UNREACHED);
+    }
+    return opResult;
+  }
+  ReaderOperandStack->push(Arg);
+
+  return 0;
+}
+
+
+IRNode *ReaderBase::generateSIMDIntrinsicCall(CORINFO_CLASS_HANDLE Class,
+  CORINFO_METHOD_HANDLE Method, CORINFO_SIG_INFO *SigInfo, ReaderBaseNS::CallOpcode Opcode) {
+
+  const char *ModuleName;
+  const char *MethodName = GetMethodName(Method, &ModuleName, JitInfo);
+  if (!strcmp(MethodName, "get_IsHardwareAccelerated")) { // Special case.
+    return generateIsHardwareAccelerated(Class);
+  } 
+  
+  uint32_t ArgsCount = SigInfo->numArgs;
+  assert(ArgsCount <= ReaderOperandStack->size());
+  if (SigInfo->hasThis()) {
+    ArgsCount += 1;
+  }
+ 
+  IRNode *ReturnNode = 0;
+
+  ReaderSIMDIntrinsic OperationType = UNDEF;
+  if (!strcmp(MethodName, ".ctor")) {
+    OperationType = CTOR;
+  }
+  else if (!strcmp(MethodName, "op_Addition")) {
+    OperationType = ADD;
+  }
+  else if (!strcmp(MethodName, "op_Subtraction")) {
+    OperationType = SUB;
+  }
+  else if (!strcmp(MethodName, "op_Multiply")) {
+    OperationType = MUL;
+  }
+  else if (!strcmp(MethodName, "op_Division")) {
+    OperationType = DIV;
+  }
+  else if (!strcmp(MethodName, "op_Equality")) {
+    OperationType = EQ;
+  }
+  else if (!strcmp(MethodName, "op_Inequality")) {
+    OperationType = NEQ;
+  }
+  else if (!strcmp(MethodName, "Abs")) {
+    OperationType = ABS;
+  }
+  else if (!strcmp(MethodName, "SquareRoot")) {
+    OperationType = SQRT;
+  }
+  else if (!strcmp(MethodName, "get_Count")) {
+    OperationType = GETCOUNTOP;
+  }
+  switch (OperationType) {
+  case ADD:
+  case SUB:
+  case MUL:
+  case DIV:
+    ReturnNode = generateSIMDBinOp(OperationType);
+    break;
+  case ABS:
+  case SQRT:
+    ReturnNode = generateSIMDUnOp(OperationType);
+    break;
+  case CTOR:
+    ReturnNode = generateSMIDCtor(Class, ArgsCount, Opcode);
+    break;
+  case GETCOUNTOP:
+    ReturnNode = vectorGetCount(Class);
+    break;
+  default:
+    ;
+  }
+  return ReturnNode;
+}
+
+IRNode *ReaderBase::generateSMIDCtor(CORINFO_CLASS_HANDLE Class, int ArgsCount, ReaderBaseNS::CallOpcode Opcode) {
+  std::vector<IRNode *> Args;
+  for (int i = 0; i < ArgsCount - 1; ++i) {
+    IRNode *Arg = ReaderOperandStack->pop();
+    Args.push_back(Arg);
+  }
+  IRNode *This = 0;
+  if (Opcode != ReaderBaseNS::CallOpcode::NewObj) {
+    This = ReaderOperandStack->pop();
+  }
+  std::reverse(Args.begin(), Args.end());
```

Comment:
In order to avoid this call to `reverse`, you could change the code that pops the arguments like so:

```
std::vector<IRNode *> Args(ArgsCount - 1);
for (int i = 0; i < ArgsCount - 1; i++)
{
  Args.push_pack(ReaderOperandStack->pop());
}